stronk/models/exercise.py

```diff
@@ -1,10 +1,37 @@
+from psycopg2.errors import UniqueViolation
+from sqlalchemy.exc import DBAPIError, IntegrityError
+from werkzeug.exceptions import BadRequest, Conflict, InternalServerError, NotFound
+
 from stronk import db
 
+
 class Exercise(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(128), index=True, nullable=False, unique=True)
     description = db.Column(db.String(500), nullable=False)
 
+    @staticmethod
+    def create(name, description):
+        exercise = Exercise(name=name, description=description)
+
+        try:
+            db.session.add(exercise)
+            db.session.commit()
+
+            return exercise
+        except IntegrityError as err:
+            raise BadRequest("Name already used by another exercise")
+        except DBAPIError as err:
+            raise InternalServerError("Database Error")
+
+    @staticmethod
+    def find_by_id(id):
+        exercise = Exercise.query.filter_by(id=id).first()
+        if not exercise:
+            raise NotFound("Exercise not found.")
```

Comment:
I think this might be the wrong `NotFound` (ie. it's 404 instead of resource error)

Reply:
True good catch